### PR TITLE
Refactor ldms and Zap libraries to use ovis_log

### PR DESCRIPTION
**Pull request by nichamon**

### CODE REVIEW

**Comment by baallan** [comment]
This is an API breaking change for all existing stream publishing clients, is it not? I see everything in our tree is being fixed, but this change takes us to a 5.x release unless there's a compatibility mode retained.

We just got a stream-dependent plugin included in a major 3rd party c++ package (kokkos), so we're going to need a compatibility API to stay in 4.x mode and not look volatile to the 3rd parties.

File: ldms/src/core/ldms_xprt.h [hunk 1]
```diff
@@ -391,8 +391,7 @@ struct ldms_xprt {
 
 	int term;
 
-	/** Transport message logging callback */
-	ldms_log_fn_t log;
+//	ldms_log_fn_t log;
```
Comment (tom95858) [inline ldms/src/core/ldms_xprt.h:394]: Please remove, let's not leave commented-out code in the tree.

Comment (nichamon) [reply ldms/src/core/ldms_xprt.h]: Done! Thanks for catching this.

**Comment by tom95858** [comment]
> This is an API breaking change for all existing stream publishing clients, is it not? I see everything in our tree is being fixed, but this change takes us to a 5.x release unless there's a compatibility mode retained.
>

The minor version speaks to binary compatibility, not API. These changes do not affect binary compatibility.
 
> We just got a stream-dependent plugin included in a major 3rd party c++ package (kokkos), so we're going to need a compatibility API to stay in 4.x mode and not look volatile to the 3rd parties.

We can discuss the merits of leaving the API the same and simply ignoring the parameter, but I think that leads to developer confusion. One out of tree plugin that we maintain doesn't seem like much of a lift.

**Pull request merged**